ssl/ssl_lib.c

```diff
@@ -5870,10 +5870,17 @@ uint64_t SSL_CTX_set_options(SSL_CTX *ctx, uint64_t op)
 
 uint64_t SSL_set_options(SSL *s, uint64_t op)
 {
-    SSL_CONNECTION *sc = SSL_CONNECTION_FROM_SSL(s);
+    SSL_CONNECTION *sc;
     OSSL_PARAM options[2], *opts = options;
 
-    if (sc == NULL)
+#ifndef OPENSSL_NO_QUIC
+    if (IS_QUIC(s) && ossl_quic_set_ssl_op(s, op))
```

Comment:
Doesn't this preclude options being set on the SSL handshake layer object...?
The code added to quic_channel.c won't be triggered either because of this also.

Reply:
Hmm... maybe just:
```suggestion
    if (IS_QUIC(s))
        (void)ossl_quic_set_ssl_op(s, op);
```

Would be OK?